lms/static/js/bookmarks/views/bookmarks_list.js

```diff
@@ -0,0 +1,124 @@
+
+            errorIcon: '<i class="fa fa-fw fa-exclamation-triangle message-error" aria-hidden="true"></i>',
+            loadingIcon: '<i class="fa fa-fw fa-spinner fa-pulse message-in-progress" aria-hidden="true"></i>',
+
+            errorMessage: gettext('An error has occurred. Please try again.'),
+            loadingMessage: gettext('Loading'),
+
+            url: '/api/bookmarks/v0/bookmarks/',
+
+            events : {
+                'click .bookmarks-results-list-item': 'visitBookmark'
+            },
+
+            initialize: function (options) {
+                this.template = _.template($('#bookmarks_list-tpl').text());
+                this.loadingMessageView = options.loadingMessageView;
+                this.errorMessageView = options.errorMessageView;
+                this.courseId = $(this.el).data('courseId');
+                this.langCode = $(this.el).data('langCode');
+                _.bindAll(this, 'render', 'breadcrumbTrail', 'userFriendlyDate', 'bookmarkUrl');
+            },
+
+            render: function () {
+                var data = {
+                    bookmarks: this.collection.models,
+                    breadcrumbTrail: this.breadcrumbTrail,
+                    userFriendlyDate: this.userFriendlyDate,
+                    bookmarkUrl: this.bookmarkUrl
+                };
+                this.$el.html(this.template(data));
+                this.delegateEvents();
+                return this;
+            },
+
+            loadBookmarks: function () {
+                var view = this;
+
+                this.hideErrorMessage();
+                this.showBookmarksContainer();
+                this.showLoadingMessage();
+
+                this.collection.url = this.url;
+                this.collection.fetch({
+                    reset: true,
+                    data: {course_id: this.courseId, fields: 'path'}
+                }).done(function () {
+                    view.hideLoadingMessage();
+                    view.render();
+                    view.focusBookmarksElement();
+                }).fail(function () {
+                    view.hideLoadingMessage();
+                    view.showErrorMessage();
+                });
+            },
+
+            visitBookmark: function (event) {
+                window.location = event.target.pathname;
+            },
+
+            breadcrumbTrail: function (bookmarkPath, unitDisplayName) {
+                var separator = ' <i class="icon fa fa-caret-right" aria-hidden="true"></i><span class="sr">-</span> ';
+                var names = _.pluck(bookmarkPath, 'display_name');
+                names.push(unitDisplayName);
+                return names.join(separator);
+            },
+
+            userFriendlyDate: function (isoDate) {
+                moment.locale(this.langCode);
+                return moment(isoDate).format('LL');
+            },
+
+            bookmarkUrl: function (courseId, usageId) {
```

Comment:
can we also rename this to `getBookmarkedUrl` or something similar because this method is being used to get or make the url for bookmarked units; correct me if i'm wrong.